tests/forms_tests/field_tests/test_datefield.py

```diff
@@ -35,7 +35,7 @@ def test_form_field(self):
 
         # label tag is correctly associated with month dropdown
         d = GetDate({'mydate_month': '1', 'mydate_day': '1', 'mydate_year': '2010'})
-        self.assertIn('<label for="id_mydate_month">', d.as_p())
+        self.assertIn('<label for="id_mydate_month"', d.as_p())
```

Comment:
I don't think that this change should be necessary. Is there something wrong in the template?

Reply:
The label now has a `\n` before the `>`. We can change to use `assertHTMLEqual` instead.

Reply:
Hmm. I think we want to avoid the `\n` in the template if we can. (Although this is inside the tag and not between tags.)

We've had weird layout quirks due to changes in whitespace when widget templates were introduced. See #9426.

Reply:
The `\n` is due to the `include` in the label templte. See [ticket 9198](https://code.djangoproject.com/ticket/9198), I'm not sure that's easily resolved. 

One thing we could do is move the `attrs.html` template into the `label.html` template as this is the only place it is used. There is also a `attrs.html` template for widgets which makes more sense as the widget templates use this frequently. Removing the form `attrs.html` template would also have the benefit of avoiding having two templates with the same name, which could get confusing.

Reply:
Ah, so we should ensure there is no trailing newline in the attrs template. This was the fix for the widget templates in 12cefee5d84b3569ccbde03167c7853e8dac638f which we should imitate for the form templates. Please add an equivalent test to ensure it doesn't regress.

Reply:
Super, thanks. I think this approach means we should be able to revert many of the `assertEqual` -> `assertHTMLEqual` changes.